### ISSUE TITLE
too few features during training

### ISSUE THREAD

**Issue by jinyummiao** [cvlab-epfl/disk#13]
Hi,

Thank you for open-sourcing your training code. 
I want to change the backbone and train a model using your training scheme. I use the default configuration. However, the model cannot extract any features during training (~300 steps) and it causes errors when calculating the "distance" and "affinity" matrix. Could you give me some advice?

Looking forward to your reply :)

**Comment by jatentaki** [comment]
Hello,

let me make sure I understood what you did. You replaced the backbone with some other neural network, ran training using default config (aside from the different backbone) and found that it crashes after ~300 optimization steps? It would be helpful to see the error itself but my guess is that you run into the problem described [here](https://github.com/cvlab-epfl/disk/blob/master/train.py#L143). Hopefully increasing `--warmup` to a higher value (say, 1000) can help remedy that. The errors in distance/affinity likely stem from those distances having a one dimension equal to 0 (which happens when there aren't any detections in one of the images).

FYI, I would interpret this as a sign that your backbone, randomly initialized, produces worse local descriptor than the UNet baseline does. It could mean that it either has a worse inductive bias or just needs more time to learn something, so not a red flag but a potentially useful diagnostic.

**Comment by jinyummiao** [comment]
Thanks for your kindly help and useful suggestions. I have increased the steps of warm up and it works.
